### ISSUE TITLE
Display the link to add the shipping address when the "No location by default" is enabled

### ISSUE THREAD

**Issue by tarunvijwani** [woocommerce/woocommerce-gutenberg-products-block#8027]
There is a setting in WooCommerce to set the default customer location. The Cart block won't show any address if there is no default value.
We need to display the link to add the shipping address when the "No location by default" is enabled:

![image](https://user-images.githubusercontent.com/11503784/209356989-894068f0-a856-4731-9aa4-043c1ab11284.png)

## Screenshots

| Actual | Expected |
|---|---|
|![image](https://user-images.githubusercontent.com/11503784/209357112-b20ba306-9641-41f3-bb0c-fdc40e02ad3a.png)| ![image](https://user-images.githubusercontent.com/11503784/209357148-e35b1523-b937-44df-8ac9-877a2264cf8b.png) |

### Slack
p1671541330197299-slack-C8X6Q7XQU

**Comment by ralucaStan** [comment]
@tarunvijwani what do you mean by default value in `The Cart block won't show any address if there is no default value.`
Also, it might be helpful to explain the behavior in the shortcode cart to better understand the issue and the expected result

**Comment by tarunvijwani** [comment]
In the [recent conversation ](p1671541330197299-slack-C8X6Q7XQU)with @elizaan36, we decided to show  "Add an address" link to make it more consistent with the "Add a coupon" link. 

>what do you mean by default value in The Cart block won't show any address if there is no default value.

If a guest user didn't add any address anywhere, it would simply show the change address link. However, the recent design shared by @elizaan36 for the coupon code had the text "Add an address for shipping rates" for it. So we created this ticket to implement it. 

>Also, it might be helpful to explain the behavior in the shortcode cart to better understand the issue and the expected result

On the cart shortcode page, the text is "Calculate shipping". However, this is an enhancement, not a bug. We want to make it more consistent with the coupon code link. 

![image](https://user-images.githubusercontent.com/11503784/210767979-cb921309-3f88-4b2c-96a3-9b0fe7cd3da2.png)
